lib/migration_tools/tasks.rb

```diff
@@ -14,60 +16,149 @@ def group
 
       @group = ENV['GROUP'].to_s
       raise "Invalid group \"#{@group}\"" if !@group.empty? && !MIGRATION_GROUPS.member?(@group)
+
       @group
     end
 
     def group=(group)
       @group = nil
-      @pending_migrations = nil
       ENV['GROUP'] = group
     end
 
     def migrations_paths
       ActiveRecord::Migrator.migrations_paths
     end
 
+    def database_configs_array
+      unless defined?(Rails) &&
+             Rails.respond_to?(:env) &&
+             ActiveRecord::Base.configurations.respond_to?(:configs_for)
+        return []
+      end
+
+      @database_configs_array ||= ActiveRecord::Base.configurations.configs_for(env_name: Rails.env)
+    end
+
+    def multi_database_setup?
+      @multi_database_setup ||= if ActiveRecord::VERSION::MAJOR >= 6
+                                  database_configs_array.reject { |c| c.spec_name.end_with?('_replica') }.size > 1
+                                else
+                                  false
+                                end
```

Comment:
instead of reject and `.size > 1` you could use `count with a block`. something like this:
```ruby
@multi_database_setup ||= ActiveRecord::VERSION::MAJOR >= 6 && database_configs_array.count { |c| !c.spec_name.end_with?('_replica') } > 1
```